rusk/src/bin/config/wallet.rs

```diff
@@ -5,72 +5,50 @@
 // Copyright (c) DUSK NETWORK. All rights reserved.
 
 use clap::{App, Arg, ArgMatches};
+use dusk_bytes::DeserializableSlice;
+use dusk_pki::PublicSpendKey;
 use rusk::Result;
 use serde::{Deserialize, Serialize};
-use std::fs;
+use tracing::warn;
 
-#[derive(Serialize, Deserialize, Clone)]
+#[derive(Serialize, Deserialize, Default, Clone)]
 pub(crate) struct WalletConfig {
-    pub(crate) path: String,
-    password: Option<String>,
-    password_path: Option<String>,
+    generator_key: Option<String>,
 }
 
-/// Default wallet path.
-pub const WALLET_PATH: &str = "wallet.dat";
-
-impl Default for WalletConfig {
-    fn default() -> Self {
-        WalletConfig {
-            path: WALLET_PATH.to_string(),
-            password: None,
-            password_path: None,
-        }
-    }
+fn parse_generator_key(key: &str) -> Result<PublicSpendKey> {
```

Comment:
I would change this function to return a `Option<PublicSpendKey>` actually, since you have it as optional and fall back to a default value in case is missing. That would simplify also other part of the code where you're using as `Option` already.

So, something like:
```rust
fn parse_generator_key(key: &str) -> Option<PublicSpendKey> {
    bs58::decode(key)
        .into_vec()
        .ok()
        .and_then(|key| PublicSpendKey::from_slice(&key).ok())
}
``` 

You won't need to import `Result` either at this point.